metricbeat/module/kubernetes/util/kubernetes.go

```diff
@@ -0,0 +1,243 @@
+package util
+
+import (
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/kubernetes"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+var nilEnricher = &enricher{}
```

Comment:
We could define a nilEnricher type that implements the interface with empty methods, to avoid needing the `if e == nilEnricher` in all methods of `enricher`.

Reply:
pushed